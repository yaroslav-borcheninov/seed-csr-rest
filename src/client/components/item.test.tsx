import React from "react"
import { shallow, ShallowWrapper } from "enzyme"

import Bootstrap from "./bootstrap"
import Item from "./item"

jest.mock("./bootstrap", () => () => null)

describe("Item", () => {
  let component: ShallowWrapper

  beforeEach(() => {
    const props = {
      match: { params: { id: "123" } },
    }
    component = shallow(<Item {...props as any} />)
  })

  it("should pass url to Bootstrap", () => {
    expect(component.find(Bootstrap).prop("url")).toEqual("/api/items/123")
  })

  it("should render data", () => {
    expect(
      component.find(Bootstrap).prop("children")({
        id: "123",
        title: "item 123",
      })
    ).toMatchSnapshot()
  })
})
