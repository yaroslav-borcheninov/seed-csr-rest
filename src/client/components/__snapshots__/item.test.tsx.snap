// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render data 1`] = `
<React.Fragment>
  <ForwardRef>
    Items
  </ForwardRef>
  <div>
    123
    : 
    item 123
  </div>
</React.Fragment>
`;
